domain/upgrade/state/state.go

```diff
@@ -40,10 +41,22 @@ func (st *State) CreateUpgrade(ctx context.Context, previousVersion, targetVersi
 	if err != nil {
 		return "", errors.Trace(err)
 	}
-	q := "INSERT INTO upgrade_info (uuid, previous_version, target_version, created_at) VALUES (?, ?, ?, DATETIME('now'))"
+	checkExistingQuery := "SELECT COUNT(*) FROM upgrade_info WHERE completed_at IS NULL"
```

Comment:
I think you could achieve this at the table level by using a unique constraint index with `IFNULL`, but this is OK.

Reply:
This would be much nicer, saving a query and reducing the length of this function dramatically but would take a bit of a contrived index. I have implemented this and left a comment explaining how it works